comms/Account.cpp

```diff
@@ -8,6 +8,78 @@
 #define ESP_EXCEPTION_LOG3(T) CReportingGSoapEspException<ns1__ArrayOfEspException> exceptions(T, __FILE__, __LINE__)
 
 #if _COMMS_VER >= 68200
+
+template <class S>
+bool CommunicateServerError(S &server, int &retCode, CString &retMsg)
+{
+    if (server.namespaces == NULL)	//  No Server  ---
+    {
+        retCode = 1003;
+        retMsg = _T("Unable to communicate with server.");
+        return true;
+    }
+    return false;
+}
+
+template <class S>
+void PasswordError(S &server, IConfig * config, int &retCode, CString &retMsg)
+{
+    _DBGLOG(config->Get(GLOBAL_SERVER_ACCOUNT), LEVEL_WARNING, server.GetClientErrorMsg());
+    retCode = server.GetClientErrorCode();
+    if (retCode == 401)
+        retMsg = _T("Invalid User ID/Password\nor password has expired");
+    else
+        retMsg = server.GetClientErrorMsg();
+}
+
+template <class R>
+bool NoErrorResponse(R &response, CString &retMsg)
+{
+    bool noExceptions = true;
+    CString responseMessage;
+#if _COMMS_VER < 300800
+    if (response.message)
+        responseMessage = response.message->c_str();
+#endif
+    if (responseMessage.IsEmpty() && response.Exceptions)
+    {
+        for (unsigned int i = 0; i < response.Exceptions->Exception.size(); ++i)
+        {
+            if (response.Exceptions->Exception[i]->Message)
+            {
+                if (!retMsg.IsEmpty())
+                    retMsg += _T("\r\n");
+
+                retMsg += response.Exceptions->Exception[i]->Message->c_str();
+                noExceptions = false;
+            }
+        }
+    }
+    return noExceptions;
+}
+
+COMMS_API bool MyAcountPasswordInfo(IConfig * config, const CString &user, const CString &password, int &retCode, CString &retMsg, int &retPasswordExpirationWarningDays, int &retPasswordDaysRemaining)
+{
+    CSoapInitialize<ws_USCOREaccountServiceSoapProxy> server(config->Get(GLOBAL_SERVER_ACCOUNT), user, password);
+    if (CommunicateServerError(server, retCode, retMsg)) return false;
+
+    _ns1__MyAccountRequest request;
+    _ns1__MyAccountResponse response;
+    ESP_EXCEPTION_LOG3(response.Exceptions);
+    if (server.MyAccount(&request, &response) == SOAP_OK)
+    {
+        SAFE_ASSIGN(retPasswordExpirationWarningDays, response.passwordExpirationWarningDays);
+        SAFE_ASSIGN(retPasswordDaysRemaining, response.passwordDaysRemaining);
+
+        return NoErrorResponse(response, retMsg);
+    }
+    else
+    {
+        PasswordError(server, config, retCode, retMsg);
+    }
+    return false;
+}
+
```

Comment:
What you had previously was cleaner than this?  All I was suggesting was moving the MyAccount call into its own function? (Note:  it doesn't need all the extra VerifyUser initialization and tear down - its just a regular SOAP call like all the other ones):
```c++
COMMS_API bool MyAccount(IConfig * config, const CString &user, const CString &password, int &retPasswordExpirationWarningDays, int &retPasswordDaysRemaining)
{
	CSoapInitialize<ws_USCOREaccountServiceSoapProxy> server(config->Get(GLOBAL_SERVER_ACCOUNT), user, password);
	_ns1__MyAccountRequest request;
	_ns1__MyAccountResponse response;
	ESP_EXCEPTION_LOG3(response.Exceptions);
	if (server.MyAccount(&request, &response) == SOAP_OK)
	{
		SAFE_ASSIGN(retPasswordExpirationWarningDays, response.passwordExpirationWarningDays);
		SAFE_ASSIGN(retPasswordDaysRemaining, response.passwordDaysRemaining);
		return true;
	}
	else
		_DBGLOG(config->Get(GLOBAL_SERVER_ACCOUNT), LEVEL_WARNING, server.GetClientErrorMsg());

	return false;
}
```